frontend/components/ifixit-image/ifixit-image.tsx

```diff
@@ -1,19 +1,51 @@
-import Image, { ImageProps, ImageLoader } from 'next/image';
+import Image, { ImageProps, ImageLoader, ImageLoaderProps } from 'next/image';
+
+interface SizeMapEntry {
+   width: number;
+   name: string;
+}
+type SizeMap = Array<SizeMapEntry>;
+
+// These should be sorted in order of ascending width.
+const guideImageSizeMap: SizeMap = [
+   { name: 'mini', width: 56 },
+   { name: 'thumbnail', width: 96 },
+   { name: '140x105', width: 140 },
+   { name: '200x150', width: 200 },
+   { name: 'standard', width: 300 },
+   { name: '440x330', width: 440 },
+   { name: 'medium', width: 592 },
+   { name: 'large', width: 800 },
+   { name: 'huge', width: 1600 },
+];
+
+const cartImageSizeMap: SizeMap = [
+   { name: 'mini', width: 41 },
+   { name: 'thumbnail', width: 70 },
+   { name: 'size110', width: 110 },
+   { name: 'size170', width: 170 },
+   { name: 'size250', width: 250 },
+   { name: 'size400', width: 400 },
+   { name: 'medium', width: 600 },
+   { name: 'size1000', width: 1000 },
+   { name: 'large', width: 3000 },
+];
 
 export function IfixitImage(props: ImageProps) {
+   let loader = props.loader;
    let unoptimized = props.unoptimized;
 
    if (typeof props.src === 'string') {
       if (isGuideImage(props.src)) {
-         unoptimized = true;
+         loader = getImageLoader(guideImageSizeMap, 'huge');
       } else if (isCartImage(props.src)) {
-         unoptimized = true;
+         loader = getImageLoader(cartImageSizeMap, 'large');
       } else if (isStrapiImage(props.src)) {
          unoptimized = true;
       }
    }
 
-   return <Image {...props} unoptimized={unoptimized} />;
+   return <Image {...props} unoptimized={unoptimized} loader={loader} />;
```

Comment:
I think setting `loader=null` here (in the case of strapi images) results in using *no* loader (instead of the default next loader). This means that strapi images are used *directly* instead of being processed by next.